hyper/hyperparamater-search.ipynb

```diff
@@ -0,0 +1,717 @@
+    "\n",
+    "Notice the use of the special `:hyperparameter:` syntax in the command body. At runtime Spell will plug the `hyperparameter` value set on the `param` flag into the corresponding `:hyperparameter:` value in the command body."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### `spell hyper random`"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 10,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "\u001b[0m💫 Casting hyperparameter search #10…\n",
+      "\u001b[0mconv2_filter    dense_layer    dropout_3    Run ID\n",
+      "60              39             0.488294     250\n",
+      "110             81             0.499862     251\n",
+      "33              40             0.34271      252\n",
+      "117             128            0.223744     253\n",
+      "117             115            0.465318     254\n",
+      "48              75             0.345887     255\n",
+      "58              121            0.292594     256\n",
+      "84              106            0.41578      257\n",
+      "104             107            0.2267       258\n",
+      "94              85             0.421738     259\n",
+      "36              99             0.399534     260\n",
+      "36              50             0.281066     261\n",
+      "30              78             0.270079     262\n",
+      "60              102            0.384012     263\n",
+      "32              115            0.498942     264\n",
+      "128             77             0.281639     265\n",
+      "19              110            0.318054     266\n",
+      "76              67             0.489551     267\n",
+      "\u001b[0m\u001b[0m"
+     ]
+    }
+   ],
+   "source": [
+    "!spell hyper random \\\n",
+    "    --machine-type K80 \\\n",
+    "    --param conv2_filter=16:128:linear:int \\\n",
+    "    --param dense_layer=32:128:linear:int \\\n",
+    "    --param dropout_3=0.2:0.5 \\\n",
+    "    --num-runs 18 \\\n",
+    "    --github-url https://github.com/spellrun/spell-examples.git \\\n",
+    "    # --github-ref hyper \\\n",
+    "    \"python hyper/cifar10_cnn.py \\\n",
+    "        --epochs 25 \\\n",
+    "        --conv2_filter :conv2_filter: \\\n",
+    "        --dense_layer :dense_layer: \\\n",
+    "        --dropout_3 :dropout_3:\""
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "This launches a random hyperparameter search. In random hyperparameter search, instead of picking a value from a list, Spell will pick a value from a range of values at random. For example, `conv2_filter` is set to the range `16:128`, so Spell might randomly pick the values `17, 125, 21, 99, 101, ...`.\n",
```

Comment:
I really like the part in your blog post about why random searches are usually better than grid searches, (I've long felt like in our mission to help guide toward 'best practices' our exposure of grid search was a bit of a sore spot), do you think its worth including a sentence or two to that effect in this notebook as well?